tests/integration/pni/search.spec.js

```diff
@@ -0,0 +1,160 @@
+const { test, expect } = require("@playwright/test");
+const waitForImagesToLoad = require("../../wait-for-images.js");
+
+/**
+ * Perform several PNI tests related to searching/filtering
+ *
+ * NOTE: this requires a `new-db` run with the seed value set
+ *       through RANDOM_SEED=530910203 in your .env file
```

Comment:
I'm thinking that this makes the integration tests quite fragile, right? Is this the reason why we have these tiny changes on Percy visual regression?

This is, of course, unrelated to this PR. I'm just thinking out loud on how we (and if) we could mitigate this issue to prevent integration tests being broken due to dev forgetting to set the `RANDOM_SEED`. I totally did not waste half an hour investigating a broken test on this today...

Reply:
Hey @jhonatan-lopes , for this I think we'll need to rely on `RANDOM_SEED` set to a fixed value. An alternative would be changing how factory work - so instead of generating a random number of products per category, we decide how many and use that value in our Playwright tests. Any thoughts on this?

Reply:
Also it looks like most of the Percy flags are due to date difference. I haven't taken a look at the code yet but I'm guessing if we instead use a fixed dates on those, then the false positive flags should go away.

Reply:
>  An alternative would be changing how factory work - so instead of generating a random number of products per category, we decide how many and use that value in our Playwright tests.

While I appreciate the randomness of it, maybe it would be best to have some controlled/reproducible situations, for instance a predictable category with 3 products that we can use in testing, similar to how we do with the "Percy blog post with a fixed title". Just an idea

Reply:
> Also it looks like most of the Percy flags are due to date difference. I haven't taken a look at the code yet but I'm guessing if we instead use fixed dates on those (or how to generate dates), then the false positive flags should go away.

That makes sense